### PR TITLE
Feature/mitigation effects tab

### PR DESCRIPTION
**Pull request by weberjavi**
[Pivotal task](https://www.pivotaltracker.com/story/show/160015512)
This PR adds the Mitigation Effects tab, under Mitigation section.
Test it on: http://localhost:3000/mitigation/mitigation-effects
The data displayed on the chart is a subset of the `table 15` of the [2nd SA BUR document](https://www.unfccc.int/files/national_reports/non-annex_i_parties/biennial_update_reports/application/pdf/south_africa_2nd_bur.pdf).

kapture 2018-08-31 at 17 32 22

TODO:
-  Define how will look the page under `SUSTAINABLE DEVELOPMENT BENEFITS` tab.
-  What's the behaviour of the `HG Emissions Reduction` filter (how it affects the chart).
-  Decide which data is to be displayed when the `Table` visualization filter is active.
-  Decide which info should be displayed on metadata modal.
-  Add bubbleChart to CW components

### CODE REVIEW

File: app/javascript/app/components/bubble-chart/bubble-chart-component.jsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import React, { Fragment, PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import ReactTooltip from 'react-tooltip';
+import { chartDataCalculation } from './bubble-chart-d3-calculation';
```
Comment (j8seangel) [inline app/javascript/app/components/bubble-chart/bubble-chart-component.jsx]: 👏

File: app/javascript/app/pages/mitigation/mitigation-effects/summary/summary-styles.scss [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  margin-top: $gutter-padding;
+  position: relative;
+}
+
+.chartContainer {
+  display: flex;
+  justify-content: center;
+
+  :global {
```
Comment (j8seangel) [inline app/javascript/app/pages/mitigation/mitigation-effects/summary/summary-styles.scss]: If we are going to reuse the tooltip styles I think worth it to move to a global scope .scss file avoiding duplicates.

**Comment by weberjavi** [comment]
@j8seangel not sure if this is what you meant when talked about a common `.scss` file. I've done it that way to have a place to handle some common resets to some libraries styles.

**Comment by Bluesmile82** [comment]
Is the info left blank on purpose as we don't have the data? Or will it be the same CAIT metadata? 
image

File: app/javascript/app/pages/mitigation/mitigation-effects/summary/summary.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        onFilterChange={this.onFilterChange}
+        updateDataColor={this.updateDataColor}
+      />
+    );
+  }
+}
+
+SummaryContainer.propTypes = {
+  updateFiltersSelected: PropTypes.func.isRequired,
+  query: PropTypes.object.isRequired
```
Comment (Bluesmile82) [inline app/javascript/app/pages/mitigation/mitigation-effects/summary/summary.js]: Could you remove the isRequired and create a defaultProp as it's not passed at the beginning?

**Comment by weberjavi** [comment]
Hey @Bluesmile82 yes I've left the Metadata blank on purpose.
I've added some TODOs on the comment which are open to have a decision on them. 
Not really sure which is the Metadata that should be showed in the modal as the data displayed on the chart has been taken from SA BUR.

**Comment by weberjavi** [comment]
@Bluesmile82 ☝️ this is ready to review again
(_are you listening slack?_)

**Pull request merged**